agent/src/main/java/com/newrelic/agent/android/NewRelic.java

```diff
@@ -1065,4 +1066,16 @@ public static boolean recordJSErrorException(StackTrace stackTrace) {
     public static boolean  addHTTPHeadersTrackingFor(List<String> headers) {
         return HttpHeaders.getInstance().addHttpHeadersAsAttributes(headers);
     }
+
+    /**
+     * Set the maximum size of the offline storage.  When the limit is reached, the agent will stop collecting offline data
+     *
+     * @param maxSize
+     */
+    public static void setMaxOfflineStorageSize(int maxSize) {
+        StatsEngine.notice().inc(MetricNames.SUPPORTABILITY_API
+                .replace(MetricNames.TAG_NAME, "setMaxOfflineStorageSize"));
+
+        OfflineStorage.setMaxOfflineStorageSize(maxSize);
```

Comment:
Is this ever enforced?

Reply:
The default value is 100MB, clients can change it up to their preference.